### PR TITLE
Spool down threads when exiting

### PR DESCRIPTION
**Pull request by jjb**

### CODE REVIEW

**Comment by jjb** [comment]
[Enhancements to stathat ruby library](https://trello.com/c/Y4E9sJEB/1825-enhancements-to-stathat-ruby-library)

File: lib/stathat.rb [hunk 1]
```diff
@@ -111,12 +110,10 @@ def ez_post_count(stat_name, ezkey, count, timestamp, cb)
...
     def run_pool
-      @runlock.synchronize { @running = true }
       @pool = []
       5.times do |i|
         @pool[i] = Thread.new do
-          while true do
-            point = @que.pop
+          while point = @que.pop do
             # XXX check for error?
             begin
```
Comment (wam) [inline lib/stathat.rb:117]: XXX check for error! :vin_diesel:

**Review by wam — APPROVED** [review]

**Review by fstutzman — APPROVED** [review]

**Comment by sidonath** [comment]
@jjb the tests seem not to be running anymore. In the previous PR I reviewed they were still working (with some warnings):

```
$ rake
/Users/damir/.rbenv/versions/2.4.2/bin/ruby -w -I"lib:lib:test" -I"/Users/damir/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/rake-12.0.0/lib" "/Users/damir/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/rake-12.0.0/lib/rake/rake_test_loader.rb" "test/test_stathat.rb"
Warning: you should require 'minitest/autorun' instead.
Warning: or add 'gem "minitest"' before 'require "minitest/autorun"'
From:
  /Users/damir/src/freedom/stathat/test/helper.rb:2:in `<top (required)>'
  /Users/damir/src/freedom/stathat/test/test_stathat.rb:1:in `<top (required)>'
  /Users/damir/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/rake-12.0.0/lib/rake/rake_test_loader.rb:15:in `block in <main>'
  /Users/damir/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/rake-12.0.0/lib/rake/rake_test_loader.rb:4:in `select'
  /Users/damir/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/rake-12.0.0/lib/rake/rake_test_loader.rb:4:in `<main>'
MiniTest::Unit.autorun is now Minitest.autorun. From /Users/damir/src/freedom/stathat/test/helper.rb:11:in `<top (required)>'
MiniTest::Unit::TestCase is now Minitest::Test. From /Users/damir/src/freedom/stathat/test/test_stathat.rb:4:in `<top (required)>'
StatHat::Reporter.instance.finish in response to EXIT
rake aborted!
Command failed with status (1): [ruby -w -I"lib:lib:test" -I"/Users/damir/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/rake-12.0.0/lib" "/Users/damir/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/rake-12.0.0/lib/rake/rake_test_loader.rb" "test/test_stathat.rb" ]

Tasks: TOP => default => test
(See full trace by running task with --trace)
```

Note the line which says "in response to EXIT".

This is the output if I change EXIT to INT in your code:

```
$ rake
/Users/damir/.rbenv/versions/2.4.2/bin/ruby -w -I"lib:lib:test" -I"/Users/damir/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/rake-12.3.0/lib" "/Users/damir/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/rake-12.3.0/lib/rake/rake_test_loader.rb" "test/test_stathat.rb"
Warning: you should require 'minitest/autorun' instead.
Warning: or add 'gem "minitest"' before 'require "minitest/autorun"'
From:
  /Users/damir/src/freedom/stathat/test/helper.rb:2:in `<top (required)>'
  /Users/damir/src/freedom/stathat/test/test_stathat.rb:1:in `<top (required)>'
  /Users/damir/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/rake-12.3.0/lib/rake/rake_test_loader.rb:17:in `block in <main>'
  /Users/damir/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/rake-12.3.0/lib/rake/rake_test_loader.rb:5:in `select'
  /Users/damir/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/rake-12.3.0/lib/rake/rake_test_loader.rb:5:in `<main>'
MiniTest::Unit.autorun is now Minitest.autorun. From /Users/damir/src/freedom/stathat/test/helper.rb:11:in `<top (required)>'
MiniTest::Unit::TestCase is now Minitest::Test. From /Users/damir/src/freedom/stathat/test/test_stathat.rb:4:in `<top (required)>'
Run options: --seed 22808

# Running:

........

Finished in 6.454818s, 1.2394 runs/s, 2.7886 assertions/s.

8 runs, 18 assertions, 0 failures, 0 errors, 0 skips
```

**Comment by jjb** [comment]
@sidonath thanks for running the tests! 

that led me down a path to discover even more improvements for the API fix branch.

**Review by sidonath — APPROVED** [review]

**Pull request merged**